market_maker_keeper/uniswapv2_market_maker_keeper.py

```diff
@@ -0,0 +1,441 @@
+
+    """
+
+    logger = logging.getLogger()
+
+    def __init__(self, args: list, **kwargs):
+        parser = argparse.ArgumentParser(prog='uniswap-market-maker-keeper')
+
+        parser.add_argument("--rpc-host", type=str, default="localhost",
+                            help="JSON-RPC host (default: `localhost')")
+
+        parser.add_argument("--rpc-port", type=int, default=8545,
+                            help="JSON-RPC port (default: `8545')")
+
+        parser.add_argument("--rpc-timeout", type=int, default=10,
+                            help="JSON-RPC timeout (in seconds, default: 10)")
+
+        parser.add_argument("--eth-from", type=str, required=True,
+                            help="Ethereum account from which to send transactions")
+
+        parser.add_argument("--eth-key", type=str, nargs='*',
+                            help="Ethereum private key(s) to use (e.g. 'key_file=aaa.json,pass_file=aaa.pass')")
+
+        parser.add_argument("--pair", type=str, required=True,
+                            help="Token pair (sell/buy) on which the keeper will operate")
+
+        parser.add_argument("--token-config", type=str, required=True,
+                            help="Token configuration file")
+
+        parser.add_argument("--uniswap-feed", type=str, required=True,
+                            help="Source of liquidity feed")
+
+        parser.add_argument("--uniswap-feed-expiry", type=int, default=86400,
+                            help="Maximum age of the liquidity feed (in seconds, default: 86400)")
+
+        parser.add_argument("--gas-price", type=int, default=9000000000,
+                            help="Gas price (in Wei)")
+
+        parser.add_argument("--smart-gas-price", dest='smart_gas_price', action='store_true',
+                            help="Use smart gas pricing strategy, based on the ethgasstation.info feed")
+
+        parser.add_argument("--initial-exchange-rate", type=float, default=None,
+                            help="Used to determine the initial ratio to be used in a newly created pool")
+
+        parser.add_argument("--percentage-difference", type=float, default=2,
+                            help="Percentage difference between Uniswap exchange rate and aggregated price"
+                                 "(default: 2)")
+
+        parser.add_argument("--uniswap-percentage-difference", type=float, default=5,
+                            help="Percentage difference between future Uniswap exchange rate and aggregated price"
+                                 "(default: 5)")
+
+        parser.add_argument("--debug", dest='debug', action='store_true',
+                            help="Enable debug output")
+
+        self.arguments = parser.parse_args(args)
+        setup_logging(self.arguments)
+
+        self.web3 = kwargs['web3'] if 'web3' in kwargs else Web3(HTTPProvider(endpoint_uri=f"{self.arguments.rpc_host}:{self.arguments.rpc_port}",
```

Comment:
can we add support for https endpoints too?

Reply:
I believe this should already support https. I've been passing https endpoints to it